pkg/plugin/validator_test.go

```diff
@@ -0,0 +1,252 @@
+				t.Errorf("Process(%+v) got unexpected validator creation error substring: %v", tc.name, diff)
+			}
+		})
+	}
+}
+
+func TestMatchIssue(t *testing.T) {
+	t.Parallel()
+
+	testPrivateKeyString, testPrivateKey := testGeneratePrivateKey(t)
+
+	cases := []struct {
+		name                    string
+		cfg                     *PluginConfig
+		issueURL                string
+		issueBytes              []byte
+		fakeTokenServerResqCode int
+		wantErrSubstr           string
+	}{
+		{
+			name:     "success",
+			issueURL: "https://github.com/test-owner/test-repo/issues/1",
+			cfg: &PluginConfig{
+				GitHubAppID:             "test-github-id",
+				GitHubAppInstallationID: "test-install-id",
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+			fakeTokenServerResqCode: http.StatusCreated,
+			issueBytes:              []byte(`{"state": "open"}`),
+		},
+		{
+			name:     "invalid_issue_url",
+			issueURL: "https://github.com/test-owner/test-repo",
+			cfg: &PluginConfig{
+				GitHubAppID:             "test-github-id",
+				GitHubAppInstallationID: "test-install-id",
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+			fakeTokenServerResqCode: http.StatusCreated,
+			wantErrSubstr:           "invalid issue url",
+			issueBytes:              []byte(`{"state": "open"}`),
+		},
+		{
+			name:     "issue_not_int",
+			issueURL: "https://github.com/test-owner/test-repo/issues/abc",
+			cfg: &PluginConfig{
+				GitHubAppID:             "test-github-id",
+				GitHubAppInstallationID: "test-install-id",
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+			fakeTokenServerResqCode: http.StatusCreated,
+			wantErrSubstr:           "invalid issue url, issueURL doesn't match pattern",
+			issueBytes:              []byte(`{"state": "open"}`),
+		},
+		{
+			name:     "unauthorized",
+			issueURL: "https://github.com/test-owner/test-repo/issues/1",
+			cfg: &PluginConfig{
+				GitHubAppID:             "test-github-id",
+				GitHubAppInstallationID: "test-install-id",
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+			fakeTokenServerResqCode: http.StatusUnauthorized,
+			wantErrSubstr:           "failed to get access token",
+			issueBytes:              []byte(`{"state": "open"}`),
+		},
+		{
+			name:     "issue_not_open",
+			issueURL: "https://github.com/test-owner/test-repo/issues/1",
+			cfg: &PluginConfig{
+				GitHubAppID:             "test-github-id",
+				GitHubAppInstallationID: "test-install-id",
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+			fakeTokenServerResqCode: http.StatusCreated,
+			wantErrSubstr:           "issue is in state: closed",
+			issueBytes:              []byte(`{"state": "closed"}`),
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		ctx := context.Background()
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			fakeTokenServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Need to close the server after the test?

Reply:
Added